sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -55,6 +55,7 @@ statement
         rowFormat?  createFileFormat? locationSpec?
         (TBLPROPERTIES tablePropertyList)?
         (AS? query)?                                                   #createTable
+    | createTableHeader LIKE source=tableIdentifier                    #createTableLike
```

Comment:
e.g. Hive only documents the simple case of `CREATE TABLE x LIKE y` without any other flags or properties
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-CreateTableLike